proto/src/code_execution.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package code_execution;
+import "google/protobuf/timestamp.proto";
+
+service TestExecution {
+    rpc QueueTest  (TestExecutionRequest) returns (TestExecutionResponse) {}
+}
+
+/*  VersionedTest represents a unique test.
+    params:
+    - test_id: ID of the test
+    - test_version: Integer representing the version of the test
+*/
+message VersionedTest {
+    int32 test_id = 1;
+    int32 test_version = 2;
+}
+
+/*  TestExecutionRequest represents a request to execute a versioned test.
+    params:
+    - session_id: ID representing the trace session
+    - versioned_test: Message representing a unique test
+    - test_input: String array representing stringified inputs to the test
+    - request_timestamp: Timestramp representing when this test execution was requested by the client
+*/
+message TestExecutionRequest {
+    int32 session_id = 1;
+    VersionedTest versioned_test = 2;
+    repeated string test_input = 3;
```

Comment:
I think we also need to support multiple parameters:

```ts
function test(input: { response: string, bar: number }, foo: boolean) {
	...
}
```

Reply:
Yup we did, slipped my mind lol. I'll change this to support varying types.

Reply:
The repeated would allow for multiple arguments unless you mean something else! 

Also toying with the idea of encoding input params as bytes and just sending over that. Then we can decode on the server side which would also decouple our supported types from our RPC calls (similar to Redis)

Reply:
> The repeated would allow for multiple arguments unless you mean something else!

Ah yeah, was thinking that you might have intended `repeated` for sending a batch of tests. Makes sense 👍

Reply:
The more I think about this, the more I feel we will eventually want to just have some common interface for encoding decoding rather than doing it in the proto. For the time being, I added some of the primitives + stringified Json. The caveat is that any array type will have to also be json stringified and parsed for now

Reply:
Discussed offline. We will just serialize using bson for all fields and send as bytes